packages/netlify-cms-widget-markdown/src/MarkdownControl/plugins/CommandsAndQueries.js

```diff
@@ -71,6 +71,37 @@ function CommandsAndQueries({ defaultType }) {
       hasInline(editor, type) {
         return editor.value.inlines.some(node => node.type === type);
       },
+      hasQuote(editor, quoteLabel) {
+        const { value } = editor;
+        const { document, blocks } = value;
+        return blocks.some(node => {
+          const { key: descendantNodeKey } = node;
+          const parent = document.getClosest(
```

Comment:
I think this has a false positive (marks a block as a quote when it shouldn't) in the following structure:
![image](https://user-images.githubusercontent.com/26760571/119351869-a80f4500-bca9-11eb-906f-3e24a99beba5.png)

Would the following code work for both use cases:
```js
return blocks.some(node => {
  const parent = document.getParent(node.key);
  return parent && parent.type == quoteLabel;
});
```

Reply:
> Hello @erezrokah . If I understand correctly:
> 
> * the first `test` is a quote block. Then you hit Enter, and the quote block continues.
> * On the the second `test`, you give it a `list` style.
> 
> Technically, the second `test` is still in quoteblock.

Correct, the thing is if you click quote block again it creates another quote block within the list item (instead of removing the root one).
We can discuss if that's the correct behavior, but as long as it is the `active` property should reflect that.

> Your suggestion works and I'm going to implement that but is it necessary to check that parent is defined? Because every thing inside the editor has a `Document` object as a parent by default. The heading block, paragraph block, their parent is the `Document`. If you give it a list or quote style, then it has a new parent. But what I mean is that the `parent` will always be defined.

You're correct. I was thinking root level objects might not have a parent, but in this context it should always be defined.


> Should I open an issue to discuss that?

What you've discovered is related to my first response in this comment. Please open a new issue for it as I also think nesting inside quote blocks it strange.